### ISSUE TITLE
When using cache to load url, check whether the cached bitmap is recycled

### ISSUE THREAD

**Issue by Saren-Arterius** [koush/ion#488]
Because I am getting exceptions loading an url into an imageview previously loaded into imagespan...

java.lang.IllegalArgumentException: Cannot draw recycled bitmaps
	at android.view.GLES20Canvas.drawBitmap(GLES20Canvas.java:791)
	at android.view.GLES20RecordingCanvas.drawBitmap(GLES20RecordingCanvas.java:118)
	at com.koushikdutta.ion.IonDrawable.draw(IonDrawable.java:707)
	at android.widget.ImageView.onDraw(ImageView.java:985)
	at android.view.View.draw(View.java:13892)
	at android.view.View.getDisplayList(View.java:12830)
	at android.view.View.getDisplayList(View.java:12874)
	at android.view.View.draw(View.java:13608)
        ...

So, basically I just want ion to invalidate the specific cache when the cached bitmap is recycled so that ion can download that image again.

**Comment by koush** [comment]
Probably a bug in your code.

5ad622b

**Comment by Saren-Arterius** [comment]
I didn't recycle bitmaps returned by ion.
I was building a material dialog and used an imageview as custom view, then I load an url into that imageview using ion. It is like there is 10% of probability that the exception will occur when after showing that material dialog.
After that I switched to activity+fragments, the problem magically disappears, I don't know why.
Anyway, thanks.

**Comment by koush** [comment]
If you can share that code, I can take a look.
